stdlib/public/libexec/swift-backtrace/TargetLinux.swift

```diff
@@ -0,0 +1,232 @@
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  Defines `Target`, which represents the process we are inspecting.
+//  This is the Linux version.
+//
+//===----------------------------------------------------------------------===//
+
+#if os(Linux)
+
+import Glibc
+
+import _Backtracing
+@_spi(Internal) import _Backtracing
+@_spi(Contexts) import _Backtracing
+@_spi(MemoryReaders) import _Backtracing
+@_spi(Utils) import _Backtracing
+
+@_implementationOnly import Runtime
+
+struct TargetThread {
+  typealias ThreadID = pid_t
+
+  var id: ThreadID
+  var context: HostContext?
+  var name: String
+  var backtrace: SymbolicatedBacktrace
+}
+
+class Target {
+  typealias Address = UInt64
+
+  var pid: pid_t
+  var name: String
+  var signal: UInt64
+  var faultAddress: Address
+  var crashingThread: TargetThread.ThreadID
+
+  var images: [Backtrace.Image] = []
+
+  var threads: [TargetThread] = []
+  var crashingThreadNdx: Int = -1
+
+  var signalName: String {
+    switch signal {
+      case UInt64(SIGQUIT): return "SIGQUIT"
+      case UInt64(SIGABRT): return "SIGABRT"
+      case UInt64(SIGBUS):  return "SIGBUS"
+      case UInt64(SIGFPE): return "SIGFPE"
+      case UInt64(SIGILL): return "SIGILL"
+      case UInt64(SIGSEGV): return "SIGSEGV"
+      case UInt64(SIGTRAP): return "SIGTRAP"
+      default: return "\(signal)"
+    }
+  }
+
+  var signalDescription: String {
+    switch signal {
+      case UInt64(SIGQUIT): return "Terminated"
+      case UInt64(SIGABRT): return "Aborted"
+      case UInt64(SIGBUS): return "Bus error"
+      case UInt64(SIGFPE): return "Floating point exception"
+      case UInt64(SIGILL): return "Illegal instruction"
+      case UInt64(SIGSEGV): return "Bad pointer dereference"
+      case UInt64(SIGTRAP): return "System trap"
+      default:
+        return "Signal \(signal)"
+    }
+  }
+
+  var reader: MemserverMemoryReader
+
+  // Get the name of a process
+  private static func getProcessName(pid: pid_t) -> String {
+    let path = "/proc/\(pid)/comm"
+    guard let name = readString(from: path) else {
+      return ""
+    }
+    return String(stripWhitespace(name))
+  }
+
+  /// Get the name of a thread
+  private func getThreadName(tid: Int64) -> String {
+    let path = "/proc/\(pid)/task/\(tid)/comm"
+    guard let name = readString(from: path) else {
+      return ""
+    }
+    let trimmed = String(stripWhitespace(name))
+
+    // Allow the main thread to use the process' name, but other
+    // threads will have an empty name unless they've set the name
+    // explicitly
+    if trimmed == self.name && pid != tid {
+      return ""
+    }
+    return trimmed
+  }
+
+  init(crashInfoAddr: UInt64, limit: Int?, top: Int, cache: Bool) {
```

Comment:
`limit` and `top` are the same as the parameters documented in the `_Backtracing` module's interface. I don't think documenting them twice is a good idea, since the documentation could then diverge. See the doc comment for `Backtrace.capture()`, as well as the comment for the `Frames.omittedFrames(count)` case

Reply:
Right, then a doc comment for this initializer could reference the source of truth. Otherwise there's nothing here and no mention of the place where this information could be looked up. Some redirection would be helpful to provide the context.